### PR TITLE
refactor(DatePicker): convert properties to camel case and deprecate those with snake case

### PR DESCRIPTION
**Pull request by joakbjerk**
- [x] Fix tests related to Fields using DatePicker

### CODE REVIEW

File: packages/dnb-eufemia/src/components/date-picker/DatePickerAddon.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { convertStringToDate } from './DatePickerCalc'
 import Button from '../button/Button'
 import DatePickerContext from './DatePickerContext'
 
+// TODO: convert shortucts properties to camelCase, constitutes a breaking change
```
Comment (langz) [inline packages/dnb-eufemia/src/components/date-picker/DatePickerAddon.tsx:11]:
```suggestion
// TODO: convert shortcut properties to camelCase, constitutes a breaking change - could be included in v11
```

File: packages/dnb-eufemia/src/components/date-picker/DatePickerDocs.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+const dateType = ['string', 'Date']
+
+export const DatePickerProperties: PropertiesTableProps = {
+  date: {
+    doc: 'Defines the pre-filled date by either a JavaScript DateInstance or (ISO 8601) like `date="2019-05-05"`` and `content`.',
```
Comment (langz) [inline packages/dnb-eufemia/src/components/date-picker/DatePickerDocs.ts:7]:
```suggestion
    doc: 'Defines the pre-filled date by either a JavaScript DateInstance or (ISO 8601) like `date="2019-05-05"` and `content`.',
```

File: packages/dnb-eufemia/src/shared/locales/nb-NO.ts [hunk 3]
```diff
@@ -22,26 +22,27 @@ export default {
...
+      returnFormat: 'yyyy-MM-dd', // used in date-fns v1: YYYY-MM-DD
+      submitButtonText: 'Ok',
+      cancelButtonText: 'Avbryt',
+      resetButtonText: 'Tilbakestill',
+      placeholderCharacters: {
         day: 'd',
         month: 'm',
         year: 'å',
       },
+      // TODO: add firstDay
```
Comment (langz) [inline packages/dnb-eufemia/src/shared/locales/nb-NO.ts:45]: Is this to be done now, or at a later stage?

**Review by langz — APPROVED** [review]

**Comment by langz** [comment]
Unit tests is probably enough to test that this change works properly. Or should I test it manually as well?

File: packages/dnb-eufemia/src/shared/locales/nb-NO.ts (hunk 3 above)
Comment (joakbjerk) [reply packages/dnb-eufemia/src/shared/locales/nb-NO.ts:45]: Thinking later stage, as it also leads to rewriting the firstDay default prop 🙏

**Comment by joakbjerk** [comment]
> Unit tests is probably enough to test that this change works properly. Or should I test it manually as well?

I think just double checking that the portal documentation looks correct should be enough? 🤔

**Comment by langz** [comment]
Is this still correct ?
https://eufemia-git-feat-date-picker-camel-case-props-eufemia.vercel.app/uilib/components/date-picker/events/

Or should it also be camelCased?

**Comment by langz** [comment]
When checking packages/dnb-design-system-portal/src/docs/uilib/components/date-picker/Examples.tsx, I see there's a good mix of props that's camelCased and other that's snake_cased, but that's probably just how it is in a transition face or so.

I find both start_date and startDate there

**Comment by langz** [comment]
Not important, but we could also update `packages/dnb-eufemia/src/components/date-picker/stories/DatePicker.stories.tsx`

**Comment by joakbjerk** [comment]
> Is this still correct ? https://eufemia-git-feat-date-picker-camel-case-props-eufemia.vercel.app/uilib/components/date-picker/events/
> 
> Or should it also be camelCased?

Should be camel cased 😝

**Comment by joakbjerk** [comment]
> When checking packages/dnb-design-system-portal/src/docs/uilib/components/date-picker/Examples.tsx, I see there's a good mix of props that's camelCased and other that's snake_cased, but that's probably just how it is in a transition face or so.
> 
> I find both start_date and startDate there

if it's snake casing in event objects, or shortcuts, then it's a symptom of the transition phase yeah, otherwise it's something that we forgot to convert 😁

**Comment by langz** [comment]
Should these changes result in the need to update a screenshot test?🤔

**Comment by joakbjerk** [comment]
> Should these changes result in the need to update a screenshot test?🤔

Not the code changes, but it seems that changing the demo examples triggered a slight change in the height in the screenshot area itself 🤷

File: packages/dnb-eufemia/src/components/date-picker/__tests__/DatePicker.test.tsx [hunk 4]
```diff
@@ -2239,8 +2231,8 @@ describe('DatePicker calc', () => {
 
   it('should inherit formElement vertical label', () => {
     render(
-      <Provider formElement={{ label_direction: 'vertical' }}>
-        <DatePicker label="Label" show_input />
+      <Provider formElement={{ labelDirection: 'vertical' }}>
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/date-picker/__tests__/DatePicker.test.tsx:2234]: Do we still support `label_direction` here? I guess, yes.

Comment (joakbjerk) [reply packages/dnb-eufemia/src/components/date-picker/__tests__/DatePicker.test.tsx:2234]: Yes we do 👍

**Review by tujoworker — APPROVED** [review]
Really nice work 🚀 

We should test this PR manually.

**Comment by langz** [comment]
In the example https://eufemia-git-feat-date-picker-camel-case-props-eufemia.vercel.app/uilib/components/date-picker/demos/#range-datepicker it's a bit annoying that we have both `startDate` and `start_date`, `endDate` and `end_date`,etc.

But I guess it's just something we need to have temporarily for now 👍

**Review by langz — APPROVED** [review]
I tested the Datepicker and Field.Date, and I could not find any anomalies

**Comment by joakbjerk** [comment]
> I tested the Datepicker and Field.Date, and I could not find any anomalies

Sweet, thank you! 🎁

**Pull request merged**